source/common/router/upstream_request.cc

```diff
@@ -204,6 +227,7 @@ class UpstreamFilterManager : public Http::FilterManager {
     state().encoder_filter_chain_aborted_ = true;
     state().remote_encode_complete_ = true;
     state().local_complete_ = true;
+    // TODO(alyssawilk) this should be done through the router to play well with hedging.
```

Comment:
Is this because we're immediately issuing a local reply here, skipping the retry/hedge handling done in the router?

Reply:
Yeah I think there is a lot of ways this could work, so probably easiest to just do something simple and then add in different mechanisms down the line if people need to also control what the retry/hedge/etc behavior should be